feat: Enhance Report Builder empty state UI

vkprogrammer-001 opened the pull request:
## Proposed Changes

- Fixes #12400 
- Added informative empty state with document icon and helpful description
- Improved visual design with dashed border and light background
- Implemented conditional UI based on user permissions
  - Users with permissions see a "Create Template" button
  - Users without permissions see guidance message
- Enhanced responsive layout for all screen sizes
- Added proper translations for all UI text

## Screenshots
<img width="1439" alt="Screenshot 2025-05-24 at 11 57 35 PM" src="https://github.com/user-attachments/assets/452712e0-13ce-4411-8ca4-de9a5599153d" />
<br>
<img width="1440" alt="Screenshot 2025-05-25 at 12 13 13 AM" src="https://github.com/user-attachments/assets/8bf3dfd6-6ba5-468b-9a13-4bb140f31de9" />


@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [x] Completion of QA in Mobile Devices
- [x] Completion of QA in Desktop Devices


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Enhanced the empty state for report templates with a styled informational panel featuring an icon, heading, and descriptive text.
  - Displayed a helpful permission notice for users without access to create templates.
  - Included new descriptive text to better explain the purpose of report templates.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

vkprogrammer-001 committed:
feat: Enhance Report Builder empty state UI

- Add informative empty state with icon and helpful description
- Improve visual design with dashed border and light background
- Implement conditional UI based on user permissions
- Show guidance for users with and without template creation permissions
- Ensure responsive layout on different screen sizes

vkprogrammer-001 committed:
Added accessibility attributes to the document icon

vkprogrammer-001 commented on src/pages/Encounters/ReportBuilder/index.tsx:
```diff
@@ -69,9 +69,72 @@ export default function ReportBuilderList({
+                        d="M10 3a1 1 0 00-1 1v5H4a1 1 0 100 2h5v5a1 1 0 102 0v-5h5a1 1 0 100-2h-5V4a1 1 0 00-1-1z"
+                        clipRule="evenodd"
+                      />
+                    </svg>
```
done

NikhilA8606 commented on src/pages/Encounters/ReportBuilder/index.tsx:
```diff
@@ -69,9 +69,74 @@ export default function ReportBuilderList({
+            <div className="text-center space-y-5 max-w-md">
+              <div className="mx-auto bg-white p-4 rounded-full w-16 h-16 flex items-center justify-center border border-gray-200 shadow-sm">
+                <svg
+                  xmlns="http://www.w3.org/2000/svg"
```
Can we use icons from Care Icons or our package instead?

vkprogrammer-001 replied:
Yes, you're right. We can use it as well. I’ve made the changes.

NikhilA8606 commented on src/pages/Encounters/ReportBuilder/index.tsx:
```diff
@@ -69,9 +69,74 @@ export default function ReportBuilderList({
+              </div>
+              <div>
+                <h4 className="text-xl font-semibold text-gray-800 mb-2">
+                  {t("no_templates_found", "No templates found")}
```
Translations should be stored in en.json file

vkprogrammer-001 replied:
done

vkprogrammer-001 committed:
Fix: icon and translation

rithviknishad commented on src/pages/Encounters/ReportBuilder/index.tsx:
```diff
@@ -69,9 +71,45 @@ export default function ReportBuilderList({
+              <div className="mx-auto bg-white p-4 rounded-full w-16 h-16 flex items-center justify-center border border-gray-200 shadow-sm">
+                <CareIcon
+                  icon="l-file-medical"
+                  className="h-8 w-8 text-blue-500 transform scale-170"
```
you can control `CareIcon`'s icon sizes using `text-*` size classes instead

```suggestion
                  className="text-blue-500 text-lg"
```

vkprogrammer-001 replied:
Thank you for the suggestion. I have implemented the changes.

rithviknishad commented on src/pages/Encounters/ReportBuilder/index.tsx:
```diff
@@ -69,9 +71,45 @@ export default function ReportBuilderList({
+                    href={`/reportbuilder/new`}
+                    className="flex items-center gap-2"
+                  >
+                    <CareIcon icon="l-plus" className="h-5 w-5" />
```
```suggestion
                    <CareIcon icon="l-plus" />
```

rithviknishad reviewed (approved):
lgtm, minor things

vkprogrammer-001 committed:
Fix: Adjust icon size using text-* size classes

abhimanyurajeesh commented on src/pages/Encounters/ReportBuilder/index.tsx:
```diff
@@ -69,9 +71,45 @@ export default function ReportBuilderList({
+          reportTemplateData?.results?.length === 0 ? (
+          <div className="flex flex-col items-center justify-center p-10 border border-dashed border-gray-300 rounded-lg bg-gray-50 my-4">
+            <div className="text-center space-y-5 max-w-md">
+              <div className="mx-auto bg-white p-4 rounded-full w-16 h-16 flex items-center justify-center border border-gray-200 shadow-sm">
```
```suggestion
              <div className="mx-auto bg-white p-4 rounded-full size-16 flex items-center justify-center border border-gray-200 shadow-sm">
```

abhimanyurajeesh commented on src/pages/Encounters/ReportBuilder/index.tsx:
```diff
@@ -69,9 +71,45 @@ export default function ReportBuilderList({
+              {canManageTemplate ? (
+                <Button
+                  variant="default"
+                  size="default"
```
defaultVariants
size is default

abhimanyurajeesh reviewed (approved):
LGTM

vkprogrammer-001 committed:
refactor(report-builder): update icon container styling and remove redundant default size prop

- Replace h-16 w-16 with size-16 utility class
- Remove explicit size="default" as it's already defined in defaultVariants

Jacobjeevan committed:
minor modifications
